packages/experiment-tag/src/experiment.ts

```diff
@@ -474,21 +491,24 @@ export class DefaultWebExperimentClient implements WebExperimentClient {
 
   private handleMutate(action, key: string, variant: Variant) {
     // Check for repeat invocations
-    if (this.appliedMutations[key]) {
+    if (this.appliedMutations[key]?.[MUTATE_ACTION]) {
       return;
     }
     const mutations = action.data?.mutations;
+    if (mutations.length === 0) {
+      return;
+    }
     const mutationControllers: MutationController[] = [];
     mutations.forEach((m) => {
       mutationControllers.push(mutate.declarative(m));
     });
-    this.appliedMutations[key] = mutationControllers;
+    (this.appliedMutations[key] ??= {})[MUTATE_ACTION] = mutationControllers;
     this.exposureWithDedupe(key, variant);
   }
 
   private handleInject(action, key: string, variant: Variant) {
     // Check for repeat invocations
-    if (this.appliedMutations[key]) {
+    if (this.appliedMutations[key]?.[INJECT_ACTION]) {
       return;
     }
```

Comment:
Do we even need this? I feel like the We already check for applied injections at line 522